alerts/machine-config-operator/MachineConfigControllerPausedPoolKubeletCA.md

```diff
@@ -0,0 +1,165 @@
+# MachineConfigControllerPausedPoolKubeletCA
+
+## Meaning
+
+The apiserver has performed a certificate rotation, but the specified
+MachineConfigPool is paused, preventing deployment of the MachineConfig
+containing the rotated  `kublet-ca.crt` bundle to the pool's nodes.
+
+>You will  need to unpause the specified pool to allow the new certificate
+>through before the existing one expires.
+
+This alert fires as a warning for a pool when:
+
+- A MachineConfigPool is paused AND
+- There is a new `kubelet-ca.crt` specified in the MachineConfigPool's spec AND
+- The pool has been in this state for more than `1 hour`
+
+This alert becomes critical for a pool when:
+
+- The above conditions are met, and the are only two weeks ( `14 days` )
+  remaining until the expiry date of the pool's most recent
+  `kube-apiserver-to-kubelet-signer` certificate.
+
+For clarity, the pool has a MachineConfig specified in
+`status.Configuration.Name`. That MachineConfig has a file inside it with a path
+of `/etc/kubernetes/kubelet-ca.crt`.  Inside that file is a certificate
+`kube-apiserver-to-kubelet-signer` which has to be valid for your nodes to work.
+The MachineConfigPool's `pause` feature is preventing that configuration from
+being updated.
+
+## Impact
+
+If the pool remains paused, the nodes in the specified pool will stop working
+when the certificates in the pool's existing `kubelet-ca.crt`  bundle expire.
+
+### Short term: warning
+
+> NOTE: This is not a desirable state to be in unless you know what you're
+> doing.
+
+After (at most) 12 hours following a certificate rotation, you will experience
+the following negative symptoms if your pool is still paused:
+
+- Pod logs for nodes in the specified pool will not be viewable in the web
+  console
+- The commands `oc logs`  ,  `oc debug` ,  `oc attach`,  `oc exec`  will not be
+  usable.
+
+This happens because:
+
+- These features depend on having a  `kubelet-client` certificate in the cluster
+  that matches the node's `kube-apiserver-to-kubelet-signer`
+- The `kubelet-client` certificate in the cluster rotates every 12 hours
+- The `kubelet-client` is signed by the *most recent*
+  `kube-apiserver-to-kubelet-signer`
+- Once a certificate rotation happens, nodes in the paused pool no longer have
+  have the most recent  `kube-apiserver-to-kubelet-signer` (it is in
+  `kubelet-ca.crt`, which is stuck behind pause)
+- So once `kubelet-client` rotates and gets signed with the most recent signer,
+  nodes in the paused pool cannot verify `kubelet-client` and therefore do not
+  trust it.
+
+Aside from these symptoms, your nodes should still be okay until
```

Comment:
Other than these symptoms, your nodes should work normally until the `kube-apiserver-to-kubelet-signer` expires.